src/main/java/com/softserve/sportshub/utils/JwtUtils.java

```diff
@@ -1,7 +1,25 @@
 package com.softserve.sportshub.utils;
 
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.JWTVerifier;
 import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.interfaces.DecodedJWT;
+
+import javax.servlet.http.HttpServletRequest;
 
 public class JwtUtils {
     public static Algorithm algorithm = Algorithm.HMAC256("secret".getBytes());
+
+    public static String getUsernameFromTokenInsideRequest (HttpServletRequest request) throws Exception {
+        String authHeader = request.getHeader("Authorization");
+        if (authHeader != null && authHeader.startsWith("Bearer ")) {
+            String refresh_token = authHeader.substring("Bearer ".length());
+            JWTVerifier verifier = JWT.require(algorithm).build();
+            DecodedJWT decodedJWT = verifier.verify(refresh_token);
+            return decodedJWT.getSubject();
+        } else {
+            throw new Exception("Authorization header is missing or invalid");
+        }
+    }
+
 }
```

Comment:
I belive this is already done in one of spring security's fillters. It is execute before the request reaches the method inside the controller, so you don't have to handle the token manually.